tools/collect-dependencies.js

```diff
@@ -0,0 +1,34 @@
+const { execSync } = require('child_process');
+const { writeFileSync } = require('fs');
+const { argv, exit } = require('process');
+
+const option = argv[3];
+const filename = argv[4];
+
+if (option !== '-o' || !filename) {
+  console.log(`command usage: ${__filename} -o filename`);
+  exit(-1);
+}
+
+console.log('Starting to collect dependencies');
+
+console.log('Invoking pnpm list...');
+const output = execSync('pnpm list -w -r --parseable --depth=0').toString('utf8');
```

Comment:
We are collecting dep=0 now?

Reply:
Agree, I would go deeper than 0, but not to Infinity.  2?